Lucky Numbers in Python

eksdee-sama opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)


### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#4772)

eksdee-sama committed:
Add files via upload

eksdee-sama committed:
Merge pull request #1 from eksdee-sama/eksdee-sama-LuckyNumbers

Add files via upload

HarshCasper reviewed (changes requested):
Add markdown updates

eksdee-sama committed:
Update README.md

eksdee-sama committed:
Update README.md

eksdee-sama committed:
Update README.md

eksdee-sama committed:
Rename Lucky Numbers.py to Lucky_Numbers.py

eksdee-sama committed:
Merge branch 'master' into master

ankitaggarwal23 reviewed (changes requested):
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

eksdee-sama committed:
Update Lucky_Numbers.py

Added Multi-Line Comments
Added a beginner-friendly description at the top
Used proper code Format
Removed extra lines of code and spaces
Added sample I/O at the end of the Code
Added Time and Space Complexity at the end of the Code

eksdee-sama committed:
Merge branch 'master' into master

HarshCasper commented on Python/math/Lucky_Numbers.py:
```diff
@@ -0,0 +1,48 @@
+original list (1, 2, 3...).
+At the point when this method has been done totally, the leftover whole numbers are the lucky numbers:
+1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99, 105, 111, 115, 127, 129, 133, 135, 141, 151, 159, 163, 169, 171, 189, 193, 195, 201, 205,
+211, 219, 223, 231, 235, 237, 241, 259, 261, 267, 273, 283, 285, 289, 297, ...)
```
Please make it short and crisp

HarshCasper commented on Python/math/Lucky_Numbers.py:
```diff
@@ -0,0 +1,48 @@
+EX2:
+input=300
+output=[1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99, 105, 111, 115, 127, 129, 133, 135, 141, 151, 159, 163, 169, 171, 189, 193, 195,
+201, 205, 211, 219, 223, 231, 235, 237, 241, 259, 261, 267, 273, 283, 285, 289, 297]
```
Use smaller I/O samples

HarshCasper commented on Python/math/Lucky_Numbers.py:
```diff
@@ -0,0 +1,48 @@
+if __name__ == "__main__":
+        n=int(input())                  #User input
+        LuckyNos=lucky(n)               #Function Call
+        print(LuckyNos)
```
use comments above the line of code not beside it

eksdee-sama committed:
Update Lucky_Numbers.py

eksdee-sama committed:
Merge branch 'master' into master

ankitaggarwal23 commented on Python/math/Lucky_Numbers.py:
```diff
@@ -0,0 +1,49 @@
+1 3 7 9 13 15 21 25
+Keep eliminating the nth leftover numbers, where n is the following number in the list after the last surviving number. Next in this model is 9. Keep repeating until the sieve
+of elimination becomes larger than the index of the number[ which is to be checked if lucky ] , then the number is called Lucky Number. If a number is eliminated by any seieve 
+then the number is not a lucky number.
```
Please make it short and readable

eksdee-sama replied:
@ankitaggarwal23 I tried to make it more readable , but really cannot make it any shorter ,as making shorter may hamper the clarity of the problem.

ankitaggarwal23 commented on Python/math/Lucky_Numbers.py:
```diff
@@ -0,0 +1,49 @@
+output=[1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99]
+EX2:
+input=150
+output=[1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99, 105, 111, 115, 127, 129, 133, 135, 141]
```
Use proper display message and write it properly

ankitaggarwal23 reviewed (changes requested):
Please make it short and readable

eksdee-sama committed:
Update Lucky_Numbers.py

-Used proper display message
-Modified the description into simpler words

eksdee-sama committed:
Merge branch 'master' into master

eksdee-sama committed:
Update Lucky_Numbers.py

ankitaggarwal23 commented on Python/math/Lucky_Numbers.py:
```diff
@@ -0,0 +1,52 @@
+The next surviving number after 7 is 9, so every 9th element was meant to be deleted, but as we see 9 is greater than size of list , So we got our list of surviving numbers 
+from seive of elimination i.e. LUCKY NUMBERS upto input number.
+
+'''
```
```
'''
Write a program to print all the Lucky Numbers upto the input number.

Approach:
Lets understand with input of 25 for example.
Start with a list of numbers beginning with 1:
1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25

Every second number (all even numbers) in the list is eliminated, leaving only the odd integers:
1 3 5 7 9 11 13 15 17 19 21 23 25

The first number remaining in the list after 1 is 3, so every third number which remains in the list 
(not every multiple of 3) is eliminated. The first of these is 5:
1 3 7 9 13 15 19 21 25

The next surviving number is now 7, so every seventh remaining number is eliminated. The first of these is 19:
1 3 7 9 13 15 21 25

The next surviving number after 7 is 9, so every 9th element was meant to be deleted, but as we see 9 is greater
than size of list , So we got our list of surviving numbers from seive of elimination i.e. LUCKY NUMBERS upto 
input number.
'''
```

Rewrite it like like this,

ankitaggarwal23 commented on Python/math/Lucky_Numbers.py:
```diff
@@ -0,0 +1,52 @@
+SAMPLE 2:-
+
+Enter a number :150
+The list of all the lucky numbers upto 130 is : [1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99, 105, 111, 115, 127, 129]
```
Remove the extra test case

eksdee-sama committed:
Update Lucky_Numbers.py
